src/qililab/platform/platform.py

```diff
@@ -983,7 +1013,10 @@ def execute_experiment(
         executor = ExperimentExecutor(
             platform=self,
             experiment=experiment,
+            database=database,
+            db_manager=self.db_manager,
```

Comment:
We don't need to pass the `db_manager` as separate parameter. It can be accessed from `platform`.

Also, based on the other comments, `database` and `optional_identifier` could be potentially removed.